batch_job/src/work_queue_factory.c

```diff
@@ -100,6 +100,8 @@ static char *wrapper_command = 0;
 static char *wrapper_input = 0;
 static char *worker_command = 0;
 
+static char *os = NULL;
```

Comment:
I suggest to rename this variable to something more explicit, e.g., runos_opsys.